cyberbrain/value_stack.py

```diff
@@ -187,6 +268,30 @@ def tos1(self):
     def tos2(self):
         return self._tos(2)
 
+    @staticmethod
+    def from_stack_items(
+        *stack_items: List[SymbolStackItem | CustomValueStackItem],
+    ) -> SymbolStackItem | CustomValueStackItem:
+        """Merges multiple SymbolStackItem into one SymbolStackItem.
+
+        If any non-SymbolStackItem is encountered, it is returned.
+        Otherwise, a new SymbolStackItem is created with the combined sources
+        and the start_lineno is the minimum of all the start_lineno's.
+        """
+        start_lineno = -1
+        sources = []
+        for item in stack_items:
+            if not isinstance(item, SymbolStackItem):
```

Comment:
Will this case actually happen when we need to merge a symbol and a custom value? If we just discard the symbol, it feels incorrect, but I could be wrong. Would be nice to show an example

Reply:
```
    try:
        assert False  # LOAD_ASSERTION_ERROR
    except AssertionError:
        pass
```
In the above code, we push an `AssertionError` onto the stack using `RAISE_VARARGS`, and then use `COMPARE_OP` to compare it with the global `AssertionError`. `COMPARE_OP` then tries to merge the two items using `merge_stack_items`. Therefore, this happens due to the fact that some op codes are used for both symbols and exceptions and if we're dealing with exceptions, the symbols should be discarded if there is any (not 100% sure on this but have yet to find a case).

Reply:
What if we return a `SymbolWithCustomValueStackItem`?

Reply:
I don't think promoting a `CustomValueStackItem` to a `SymbolWithCustomValueStackItem` would make sense, even when combined with other `SymbolStackItems`, since all relevant items are only used for exception handling, correct me if I'm wrong. The only other case would be merging `SymbolWithCustomValueStackItem` with `SymbolStackItems`, to return a new `SymbolWithCustomValueStackItem`, which does make sense, but I was thinking of pushing it to a later PR to include other kinds of custom/symbol items since this is new functionality and this PR is already big as is.

Reply:
You brought up a very good point that we assume custom values are only used in error handling, which is usually correct. But exceptions can just be a "regular" value. I filed #147. Fine with treat it later. No need to change the code.

Reply:
In those cases, we'lll still be using `SymbolWithCustomValueStackItem` to handle those cases instead of `CustomValueStackItem`

Reply:
SGTM